pkg/auth/authorizer/abac/abac.go

```diff
@@ -52,7 +53,46 @@ func (p policyLoadError) Error() string {
 type policyList []*api.Policy
 
 // TODO: Have policies be created via an API call and stored in REST storage.
-func NewFromFile(path string) (policyList, error) {
+func NewFromFile(path string) (*policyList, error) {
+
+	pl, err := newFromFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an empty watcher
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch the policy file
+	err = watcher.Add(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// On write operations on the policy file, reload policy file
+	go func() {
+		for {
+			select {
+			case ev := <-watcher.Events:
+				switch ev.Op {
+				case fsnotify.Write:
```

Comment:
if a file is deleted, it's no longer watched. It uses inotify under the cover, so this is a limitation of inotify. Renaming is essentially a copy+delete. Probably the only way around this problem is by polling, which is kind of ugly. If we're adding a new flag per above comment, perhaps this limitation can be clearly stated there.